gnomad/utils/constraint.py

```diff
@@ -700,3 +704,207 @@ def get_all_pop_lengths(
     )
 
     return pop_lengths
+
+
+def explode_by_vep_annotation(
+    t: Union[hl.Table, hl.MatrixTable],
+    vep_annotation: str,
+    vep_root: str = "vep",
+) -> Union[hl.Table, hl.MatrixTable]:
+    """
+    Annotate specified VEP annotation if it's not in the input Table or MatrixTable and explode the VEP annotation.
+
+    :param t: Input Table or MatrixTable.
+    :param vep_annotation: Name of annotation in VEP annotation that will be annotated,
+        if necessary, and explode.
+    :param vep_root: Name used for VEP annotation. Default is 'vep'.
+    :return: Table or MatrixTable with exploded VEP annotation.
+    """
+    # Annotate 'worst_csq_by_gene' to t if it's specified for `vep_annotation`.
+    if vep_annotation == "worst_csq_by_gene":
+        t = process_consequences(t)
+
+    if vep_annotation not in t[vep_root].keys():
+        raise ValueError(
+            f"{vep_annotation} is not a row field of the VEP annotation in"
+            " Table/MatrixTable!"
+        )
+    # Create top-level annotation for `vep_annotation` and explode it.
+    if isinstance(t, hl.Table):
+        t = t.transmute(**{vep_annotation: t[vep_root][vep_annotation]})
+        t = t.explode(t[vep_annotation])
+    else:
+        t = t.transmute_rows(**{vep_annotation: t[vep_root][vep_annotation]})
+        t = t.explode_rows(t[vep_annotation])
+
+    return t
+
+
+def get_constraint_grouping_expr(
+    vep_annotation_expr: hl.StructExpression,
+    coverage_expr: hl.Int32Expression = None,
+    include_transcript_group: bool = True,
+    include_canonical_group: bool = True,
+) -> Dict[str, Union[hl.StringExpression, hl.Int32Expression, hl.BooleanExpression]]:
+    """
+    Collect annotations used for constraint groupings.
+
+    Function collects the following annotations:
+        - annotation - 'most_severe_consequence' annotation in `vep_annotation`
+        - modifier - classic lof annotation, LOFTEE annotation, or PolyPhen annotation
+          in `vep_annotation`
```

Comment:
Or "None" if neither is defined. Also specify that the first two are pulled from the 'lof' annotation on `vep_annotation_expr` and the last is from the 'polyphen_prediction' annotation on `vep_annotation_expr`

`vep_annotation` no longer exists